### ISSUE TITLE
Non-existent aliases tripping up arrays()?

### ISSUE THREAD

**Issue by adavidzh** [scikit-hep/uproot4#61]
I took `uproot4` for a spin for the first time and it choked on a call to `arrays()`:
```console
---------------------------------------------------------------------------
KeyInFileError                            Traceback (most recent call last)
~/workspace/hgc_l1_trigger_autoencoder/reco/reco.py in 
----> 24 hits.arrays()

/usr/local/lib/python3.8/site-packages/uproot4/behaviors/TBranch.py in arrays(self, expressions, cut, filter_name, filter_typename, filter_branch, aliases, compute, entry_start, entry_stop, decompression_executor, interpretation_executor, array_cache, library, how)
    914                 return None
    915 
--> 916         aliases = _regularize_aliases(self, aliases)
    917         arrays, expression_context, branchid_interpretation = _regularize_expressions(
    918             self,

/usr/local/lib/python3.8/site-packages/uproot4/behaviors/TBranch.py in _regularize_aliases(hasbranches, aliases)
    100 def _regularize_aliases(hasbranches, aliases):
    101     if aliases is None:
--> 102         return hasbranches.aliases
    103     else:
    104         new_aliases = dict(hasbranches.aliases)

/usr/local/lib/python3.8/site-packages/uproot4/behaviors/TTree.py in aliases(self)
     21     @property
     22     def aliases(self):
---> 23         aliases = self.member("fAliases")
     24         if aliases is None:
     25             return {}

/usr/local/lib/python3.8/site-packages/uproot4/model.py in member(self, name, bases, recursive_bases, none_if_missing)
    288             return None
    289         else:
--> 290             raise uproot4.KeyInFileError(
    291                 name,
    292                 """{0}.{1} has only the following members:

KeyInFileError: not found: 'fAliases' because .Model_TTree_v5 has only the following members:

    '@fUniqueID', '@fBits', 'fName', 'fTitle', 'fLineColor', 'fLineStyle', 'fLineWidth', 'fFillColor', 'fFillStyle', 'fMarkerColor', 'fMarkerStyle', 'fMarkerSize', 'fEntries', 'fTotBytes', 'fZipBytes', 'fSavedBytes', 'fTimerInterval', 'fScanField', 'fUpdate', 'fMaxEntryLoop', 'fMaxVirtualSize', 'fAutoSave', 'fEstimate', 'fBranches', 'fLeaves', 'fIndexValues', 'fIndex'

in file ../dat/config1_mu+_95_to_100GeV_4Tesla_t0.root
```

The tree looks like:
```console

hits.show(interpretation_width=50)
name                 | typename             | interpretation                                    
---------------------+----------------------+---------------------------------------------------
eventID              | int32_t              | AsDtype('>i4')                                    
pdgID                | int32_t              | AsDtype('>i4')                                    
totalEnergy_GeV      | double               | AsDtype('>f8')                                    
P_X_GeV              | double               | AsDtype('>f8')                                    
P_Y_GeV              | double               | AsDtype('>f8')                                    
P_Z_GeV              | double               | AsDtype('>f8')                                    
beamX_cm             | double               | AsDtype('>f8')                                    
beamY_cm             | double               | AsDtype('>f8')                                    
beamZ_cm             | double               | AsDtype('>f8')                                    
hit_ID               | std::vector<int32_t> | AsJagged(AsDtype('>i4'), header_bytes=10)         
hit_x_cm             | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_y_cm             | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_z_cm             | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_Edep_keV         | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_EdepNonIonizing_ | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_TOA_ns           | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_TOA_last_ns      | std::vector<double>  | AsJagged(AsDtype('>f8'), header_bytes=10)         
hit_type             | std::vector<int32_t> | AsJagged(AsDtype('>i4'), header_bytes=10)         
ESum_MeV             | double               | AsDtype('>f8')                                    
COG_Z_cm             | double               | AsDtype('>f8')                                    
NHits                | int32_t              | AsDtype('>i4')
```

One fishy thing is that any call of the form `hist[col_name].array()` is perfectly fine, but `hist.arrays([col1, col2])` fails.

So, I tried to make a MWE, but unfortunately it fails to reproduce so I wonder about the root of the issue:
```python
#%%
from ROOT import TFile, TTree
from array import array

f = TFile('foo.root','recreate')
t = TTree('foo','foo')
maxn = 10
n = array('i',[0])
d = array('f',maxn*[0.])
t.Branch('mynum',n,'mynum/I')
t.Branch('myval',d,'myval[mynum]/D')

for i in range(25):
    n[0] = min(i,maxn)
    for j in range(n[0]):
        d[j] = i*0.1+j
        t.Fill()

f.Write()
f.Close()

# %%
import uproot4 as ur

#%%
t = ur.open('./foo.root:foo')

# %%
t.show()

# %%
t.arrays(library='pd')
```

This example's `show()` yields:
```console
name                 | typename             | interpretation                    
---------------------+----------------------+-----------------------------------
mynum                | int32_t              | AsDtype('>i4')                    
myval                | double[]             | AsJagged(AsDtype('>f8'))
```
 
I've noticed a couple of differences between the failing and passing cases:
- Failing is coming from C++, while passing is all Python,
- Failing uses `std::vector<>`, while passing uses `array`,
- Failing prints `AsJagged(AsDtype('>f8'), header_bytes=10) `, while passing has just `AsJagged(AsDtype('>f8'))`, i.e. no `header_bytes=10`.

Let me know if any of this is helpful and whether I can help making better sense of it.

**Comment by jpivarski** [comment]
> ```
> /usr/local/lib/python3.8/site-packages/uproot4/behaviors/TTree.py in aliases(self)
>      21     @property
>      22     def aliases(self):
> ---> 23         aliases = self.member("fAliases")
>      24         if aliases is None:
>      25             return {}
> ```

At that point in the code, I must have thought that `self.member(..., none_if_missing=True)` was the default.

> `KeyInFileError: not found: 'fAliases' because .Model_TTree_v5 has only`

The earliest version of ROOT that I use for testing is 5.24/00, which is about 5 years old. The TTree version in that ROOT version is 16. Your TTree is version 5! That's why it doesn't have some members, like `fAliases`, but that's also a good test. (Did it come from GEANT? GEANT has an independent ROOT-writer, which generates some very old versions of common ROOT objects.)

Since this ought to be a simple fix, I committed it directly as d67d431. If your problem isn't fixed, either reopen issue (if possible) or comment here (I'll see it, either way).

**Comment by adavidzh** [comment]
> Your TTree is version 5! That's why it doesn't have some members, like `fAliases`, but that's also a good test. 

I did notice the `v5` and the `dict` that starts in 16 but didn't think too much of it.

>(Did it come from GEANT? GEANT has an independent ROOT-writer, which generates some very old versions of common ROOT objects.)

Yes, it's coming from GEANT; 💯 points for that alone.

> Since this ought to be a simple fix, I committed it directly as [d67d431](https://github.com/scikit-hep/uproot4/commit/d67d431483e7b2b38d01305311438b2a5685566d). If your problem isn't fixed, either reopen issue (if possible) or comment here (I'll see it, either way).

Tested to be working. 🥳

Thanks!
